packages/react-native-web/src/types/styles.js

```diff
@@ -0,0 +1,280 @@
+  transitionProperty?: string | Array<string>,
+  transitionTimingFunction?: string | Array<string>
+|};
+
+/**
+ * Border
+ */
+
+type BorderRadiusValue = number | string;
+type BorderStyleValue = 'solid' | 'dotted' | 'dashed';
+
+export type BorderStyles = {|
+  borderColor?: ColorValue,
+  borderBottomColor?: ColorValue,
+  borderEndColor?: ColorValue,
+  borderLeftColor?: ColorValue,
+  borderRightColor?: ColorValue,
+  borderStartColor?: ColorValue,
+  borderTopColor?: ColorValue,
+  borderRadius?: BorderRadiusValue,
+  borderBottomEndRadius?: BorderRadiusValue,
+  borderBottomLeftRadius?: BorderRadiusValue,
+  borderBottomRightRadius?: BorderRadiusValue,
+  borderBottomStartRadius?: BorderRadiusValue,
+  borderTopEndRadius?: BorderRadiusValue,
+  borderTopLeftRadius?: BorderRadiusValue,
+  borderTopRightRadius?: BorderRadiusValue,
+  borderTopStartRadius?: BorderRadiusValue,
+  borderStyle?: BorderStyleValue,
+  borderBottomStyle?: BorderStyleValue,
+  borderEndStyle?: BorderStyleValue,
+  borderLeftStyle?: BorderStyleValue,
+  borderRightStyle?: BorderStyleValue,
+  borderStartStyle?: BorderStyleValue,
+  borderTopStyle?: BorderStyleValue
+|};
+
+/**
+ * Interactions
+ */
+
+type CursorValue =
+  | 'alias'
+  | 'all-scroll'
+  | 'auto'
+  | 'cell'
+  | 'context-menu'
+  | 'copy'
+  | 'crosshair'
+  | 'default'
+  | 'grab'
+  | 'grabbing'
+  | 'help'
+  | 'pointer'
+  | 'progress'
+  | 'wait'
+  | 'text'
+  | 'vertical-text'
+  | 'move'
+  | 'none'
+  | 'no-drop'
+  | 'not-allowed'
+  | 'zoom-in'
+  | 'zoom-out'
+  // resize
+  | 'col-resize'
+  | 'e-resize'
+  | 'ew-resize'
+  | 'n-resize'
+  | 'ne-resize'
+  | 'ns-resize'
+  | 'nw-resize'
+  | 'row-resize'
+  | 's-resize'
+  | 'se-resize'
+  | 'sw-resize'
+  | 'w-resize'
+  | 'nesw-resize'
+  | 'nwse-resize';
+
+type TouchActionValue =
+  | 'auto'
+  | 'inherit'
+  | 'manipulation'
+  | 'none'
+  | 'pan-down'
+  | 'pan-left'
+  | 'pan-right'
+  | 'pan-up'
+  | 'pan-x'
+  | 'pan-y'
+  | 'pinch-zoom';
+
+type UserSelect = 'all' | 'auto' | 'contain' | 'none' | 'text';
+
+export type InteractionStyles = {|
+  // https://developer.mozilla.org/en-US/docs/Web/CSS/cursor#Formal_syntax
+  cursor?: CursorValue,
+  // https://developer.mozilla.org/en-US/docs/Web/CSS/touch-action#Formal_syntax
+  touchAction?: TouchActionValue,
+  // https://developer.mozilla.org/en-US/docs/Web/CSS/user-select#Formal_syntax_2
+  userSelect?: UserSelect,
+  willChange?: string
+|};
+
+/**
+ * Layout
+ */
+
+type OverflowValue = 'auto' | 'hidden' | 'scroll' | 'visible';
+type VisiblilityValue = 'hidden' | 'visible';
+
+export type LayoutStyles = {|
+  alignContent?:
+    | 'center'
+    | 'flex-end'
+    | 'flex-start'
+    | 'space-around'
+    | 'space-between'
+    | 'stretch',
+  alignItems?: 'baseline' | 'center' | 'flex-end' | 'flex-start' | 'stretch',
+  alignSelf?: 'auto' | 'baseline' | 'center' | 'flex-end' | 'flex-start' | 'stretch',
+  backfaceVisibility?: VisiblilityValue,
+  borderWidth?: NumberOrString,
```

Comment:
Change these types to `DimensionValue`: https://github.com/facebook/react-native/blob/aee88b6843cea63d6aa0b5879ad6ef9da4701846/Libraries/StyleSheet/StyleSheetTypes.js#L27